internal/mail/message.go

```diff
@@ -0,0 +1,142 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information
+
+package mail
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"mime"
+	"mime/multipart"
+	"mime/quotedprintable"
+	"net/textproto"
+	"strings"
+	"time"
+)
+
+// Message is RFC compliant email message
+type Message struct {
+	From      Address
+	To        []Address
+	Subject   string
+	ID        string
+	Date      time.Time
+	ReceiptTo []string
+
+	ContentType string
+	Encoding    string
+	Disposition string
+
+	Body  string
+	Parts []Part
+}
+
+// Address is email address of a sender/recipient with name
+type Address struct {
+	Email string
+	Name  string
+}
+
+// Part represent one part of multipart message
+type Part struct {
+	Type        string
+	Encoding    string
+	Disposition string
+	Content     string
+}
+
+// Bytes builds message and returns result as bytes
+func (msg *Message) Bytes() ([]byte, error) {
+	// always returns nil on read and write, so most of the errors can be ignored
+	var body bytes.Buffer
+
+	// write headers
+	fmt.Fprintf(&body, "MIME-Version: 1.0\r\n")
+	fmt.Fprintf(&body, "Subject: %v\r\n", msg.Subject)
+	fmt.Fprintf(&body, "From: %s\r\n", formatAddress(msg.From))
+	for _, to := range msg.To {
+		fmt.Fprintf(&body, "To: %s\r\n", formatAddress(to))
+	}
+	for _, recipient := range msg.ReceiptTo {
+		fmt.Fprintf(&body, "Disposition-Notification-To: <%v>\r\n", recipient)
+	}
+	// date and id are optional as they can be set by server itself
+	if !msg.Date.IsZero() {
+		fmt.Fprintf(&body, "Date: %v\r\n", msg.Date)
+	}
+	if msg.ID != "" {
+		fmt.Fprintf(&body, "Message-ID: <%v>\r\n", msg.ID)
+	}
+
+	// parse content type
+	mtype, _, err := mime.ParseMediaType(msg.ContentType)
```

Comment:
Can we make a requirement that Body is always utf-8 plain text, maybe rename `Body` -> `PlainText` as well.